examples/reinforcement_learning/ddpg_pendulum.py

```diff
@@ -0,0 +1,218 @@
+    done = xp.asarray([sample[3] for sample in samples], dtype=np.float32)
+    s_next = xp.asarray([sample[4] for sample in samples], dtype=np.float32)
+
+    def update_Q():
+        # Predicted values: Q(s,a)
+        y = F.reshape(Q(s, a), (n,))
+        # Target values: r + gamma * Q(s,policy(s))
+        with chainer.no_backprop_mode():
+            next_q = F.reshape(target_Q(s_next, target_policy(s_next)), (n,))
+            t = r + gamma * (1 - done) * next_q
+        loss = F.mean_squared_error(y, t)
+        Q.cleargrads()
+        loss.backward()
+        opt_Q.update()
+
+    def update_policy():
+        # Maximize Q(s,policy(s))
+        loss = - F.sum(Q(s, policy(s))) / n
+        policy.cleargrads()
+        loss.backward()
+        opt_policy.update()
+
+    update_Q()
+    update_policy()
+
+
+def soft_copy_params(source, target, tau):
+    for s, t in zip(source.params(), target.params()):
+        t.data[:] += tau * (s.data - t.data)
+
+
+def main():
+
+    parser = argparse.ArgumentParser(description='Chainer example: DRL(DDPG)')
+    parser.add_argument('--env', type=str, default='Pendulum-v0',
+                        help='Name of the OpenAI Gym environment to play')
+    parser.add_argument('--batchsize', '-b', type=int, default=64,
+                        help='Number of transitions in each mini-batch')
+    parser.add_argument('--episodes', '-e', type=int, default=1000,
+                        help='Number of episodes to run')
+    parser.add_argument('--gpu', '-g', type=int, default=-1,
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='ddpg_result',
+                        help='Directory to output the result')
+    parser.add_argument('--unit', '-u', type=int, default=100,
+                        help='Number of units')
+    parser.add_argument('--reward-scale', type=float, default=1e-3,
+                        help='Reward scale factor')
+    parser.add_argument('--replay-start-size', type=int, default=500,
+                        help='Number of steps after which replay is started')
+    parser.add_argument('--tau', type=float, default=1e-2,
+                        help='Softness of soft target update')
+    parser.add_argument('--noise-scale', type=float, default=0.4,
+                        help='Scale of additive Gaussian noises')
+    parser.add_argument('--record', action='store_true', default=True,
+                        help='Record performance')
+    parser.add_argument('--no-record', action='store_false', dest='record')
+    args = parser.parse_args()
+
+    # Initialize an environment
+    env = gym.make(args.env)
+    assert isinstance(env.observation_space, gym.spaces.Box)
+    assert isinstance(env.action_space, gym.spaces.Box)
+    ndim_obs = env.observation_space.low.size
+    ndim_action = env.action_space.low.size
+    if args.record:
+        env.monitor.start(args.out, force=True)
```

Comment:
Due to deprecation of env.monitor, this has to be changed to `env = gym.wrappers.Monitor(env, args.out, force=True)`. I checked and it works.